test/vesting.test.js

```diff
@@ -0,0 +1,130 @@
+const { expect } = require("chai")
+const { ethers } = require("hardhat")
+const { BigNumber } = require("ethers");
+
+describe.only("Vesting", function () {
+    let owner
+    let acc2
+    let acc3
+    let acc4
+    let vesting
+    let DPT
+    beforeEach(async function () {
+        const DniproToken = await ethers.getContractFactory("DPT");
+        DPT = await DniproToken.deploy("DniproToken", "DPT", BigNumber.from("100000000").mul(BigNumber.from("10").pow(18)));
+        [owner, acc2, acc3, acc4] = await ethers.getSigners()
+        const Vesting = await ethers.getContractFactory("Vesting", owner)
+        vesting = await Vesting.deploy(DPT.address)
+    })
+
+    it("Vesting started successfully!", async function () {
+        const amountOfDistribution = 10000;
+        const periodOfDistributionInDays = 7;
+        const amountOfDistributionPerUser = 100;
+        await DPT.approve(vesting.address, amountOfDistribution)
+        await expect(vesting.connect(acc2).claimRevard()).to.be.revertedWith("Vesting has not started");
+        await expect(vesting.connect(acc2).join()).to.be.revertedWith("Vesting has not started");
+        await vesting.connect(owner).startVesting(amountOfDistribution, periodOfDistributionInDays, amountOfDistributionPerUser)
+        await expect(vesting.connect(acc2).claimRevard()).to.be.revertedWith("You have not joined the vesting");
+        expect(await DPT.balanceOf(vesting.address)).to.eq(10000)
+        expect(await vesting.maxUsers()).to.eq(100)
+        await vesting.connect(acc2).join()
+        expect(await vesting.users()).to.eq(1)
+    })
+    it("user should joined vesting", async function () {
```

Comment:
test case messages not representing the true things that happens in ur tests, try better